File: 189-rotate-array/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = int(len(nums))
        k = k % l
        split_val = l - k
        new_sublist = nums[0:split_val]
        nums[0:split_val] = []
        nums.extend(new_sublist)
        

File: 189-rotate-array/rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    let l = nums.length;
    k = k % l;
    
    const reverse = (arr, start, end) => {
        while(start  < end){
        let temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
        start++;
        end--;};
    };

    reverse(nums, 0,l-1);
    reverse(nums, 0,k-1);
    reverse(nums,k ,l-1);

    console.log(nums);

};